File: tools/src/export/roster-json.ts
/**
 * Canonical Roster JSON serializer — emits the {@link Roster} as 2-space JSON,
 * the same shape the importers consume. This is the lossless pivot, so the
 * pretty-printed text is exactly `roster.schema.json` shape.
 *
 * @packageDocumentation
 */
import type { Roster } from "../import/types.js";
import { prettyJson } from "./helpers.js";
import type { RosterSerializer } from "./serializer.js";

export const rosterJsonSerializer: RosterSerializer = {
  id: "roster-json",
  serialize(roster: Roster): string {
    return prettyJson(roster);
  },
};

File: crates/wh40kdc/src/export/roster_json.rs
//! Canonical Roster JSON serializer — emits the [`Roster`] as 2-space JSON,
//! the same shape the importers consume. This is the lossless pivot, so the
//! pretty-printed text is exactly `roster.schema.json` shape.
//!
//! Rust mirror of `tools/src/export/roster-json.ts`.

use crate::import::Roster;

use super::helpers::pretty_json;
use super::{ExportFormat, RosterSerializer};

pub struct RosterJsonSerializer;

impl RosterSerializer for RosterJsonSerializer {
    fn id(&self) -> ExportFormat {
        ExportFormat::RosterJson
    }

    fn serialize(&self, roster: &Roster) -> String {
        pretty_json(roster)
    }
}
